### PR TITLE
Sq lbranch

### PR DESCRIPTION
**Pull request by AndreyDMA**

### CODE REVIEW

File: src/main/java/mateacademy/internetshop/Factory.java [hunk 1]
```diff
@@ -16,73 +20,54 @@
...
+        } catch (ClassNotFoundException | SQLException e) {
+            logger.error("Connection to our DB was not provided");
+        }
+    }
 
     private Factory() {
     }
 
     public static ItemDao getItemDao() {
-        if (instanceItemDao == null) {
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/Factory.java:34]: if (instanceItemDao == null) {
            instanceItemDao = new ItemDaoJdbcImpl(connection);
        }
        return instanceItemDao;

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    @Override
+    public List<Item> getAll() {
+        return null;
+    }
+
+    @Override
+    public Item create(Item item) {
+        Statement stmt = null;
+        String query = "INSERT INTO " + DB_NAME + ".items (name, price) "
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: we can use String.format() in all queries

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    @Override
+    public List<Item> getAll() {
+        return null;
+    }
+
+    @Override
+    public Item create(Item item) {
+        Statement stmt = null;
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]:
```suggestion
        Statement statement = null;
```

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        String query = "INSERT INTO " + DB_NAME + ".items (name, price) "
+                + "VALUES ('" + item.getName() + "', '" + item.getPrice() + "');";
+        try {
+            stmt = connection.createStatement();
+            stmt.executeUpdate(query);
+            return item;
+        } catch (SQLException e) {
+            logger.warn("Can't create item " + item.getName());
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]:
```suggestion
            logger.error("Can't create item " + item.getName());
```

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        } finally {
+            if (stmt != null) {
+                try {
+                    stmt.close();
+                } catch (SQLException e) {
+                    logger.warn("Can't close statement ", e);
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]:
```suggestion
                    logger. error("Can't close statement ", e);
```

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        return null;
+    }
+
+    @Override
+    public Item get(Long id) {
+        Statement stmt = null;
+        String query = "SELECT * FROM " + DB_NAME + ".items where item_id = " + id + ";";
+        try {
+            stmt = connection.createStatement();
+            ResultSet rs = stmt.executeQuery(query);
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]:
```suggestion
            ResultSet resultSet = stmt.executeQuery(query);
```

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        }
+        return null;
+    }
+
+    @Override
+    public Item update(Item item) {
+        Statement stmt = null;
+        String query = "UPDATE " + DB_NAME + ".items SET name = '"
+                + item.getName() + "', price = " + item.getPrice()
+                + " WHERE (item_id = " + item.getItemId() + ");";
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]:
```suggestion
                + " WHERE item_id = " + item.getItemId() + ";";
```

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void delete(Long itemId) {
+        Statement stmt = null;
+        String query = "DELETE FROM " + DB_NAME
+                + ".items WHERE (item_id = " + itemId + ");";
```
Comment (Sofasmile) [inline src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]:
```suggestion
                + ".items WHERE item_id = " + itemId + ";";
```

File: src/main/java/mateacademy/internetshop/Factory.java (hunk 1 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/Factory.java:34]: fixed

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 2 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: done

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 3 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: done

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 4 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: fixed

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 5 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: fixed

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 6 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: done

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 7 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: fixed

File: src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 8 above)
Comment (AndreyDMA) [reply src/main/java/mateacademy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: fixed

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**